app/services/voucher_adjustments_service.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+class VoucherAdjustmentsService
+  def self.calculate(order)
+    return if order.nil?
+
+    # Find open Voucher Adjustment
+    return if order.voucher_adjustments.empty?
+
+    # We only support one voucher per order right now, we could just loop on voucher_adjustments
+    adjustment = order.voucher_adjustments.first
+
+    # Recalculate value
+    amount = adjustment.originator.compute_amount(order)
+
+    # It is quite possible to have an order with both tax included in and tax excluded from price.
+    # We should be able to caculate the relevant amount apply the current calculation.
+    #
+    # For now we just assume it is either all tax included in price or all tax excluded from price.
```

Comment:
This seems faulty, and so I feel cautious about introducing it. But if everyone's aware and it's been documented in other issues, all good.

Reply:
Currently, we don't know of a single instance which mixes included and excluded tax. Canada will probably be the first when they start to fully support different alcohol taxes in the regions but as of now, this "fault" doesn't matter.